src/lib/utils/index.ts

```diff
@@ -217,3 +219,28 @@ export const timeout = <Left, Right>(
     ),
   ])
 }
+
+export const getSupportedChainIdsByChainNamespace = () => {
+  return Array.from(getChainAdapterManager().keys()).reduce<Record<ChainNamespace, ChainId[]>>(
+    (acc, chainId) => {
+      const { chainNamespace } = fromChainId(chainId)
+      if (!acc[chainNamespace]) acc[chainNamespace] = []
+      acc[chainNamespace].push(chainId)
+      return acc
+    },
+    {} as Record<ChainNamespace, ChainId[]>,
+  )
+}
+
+export const assertGetChainAdapter = (
+  chainId: ChainId | KnownChainIds,
+): ChainAdapter<KnownChainIds> => {
+  const chainAdapterManager = getChainAdapterManager()
+  const adapter = chainAdapterManager.get(chainId)
+
+  if (adapter === undefined) {
+    throw Error(`chain adapter not found for chain id ${chainId}`)
+  }
+
+  return adapter as unknown as ChainAdapter<KnownChainIds>
```

Comment:
lol, this almost feels like sweeping dust under the rug. We could add an assertion function to me a little more idiomatic, e.g. of the format:

```ts
function assertIsString(val: any): asserts val is string {
  if (typeof val !== "string") {
    throw new AssertionError("Not a string!");
  }
}
```

Reply:
ah i've pushed a fix for that, no longer required :) 30f08f976539488c8410a5669c7ccdc7600fb9fa